kernel/tests/read.rs

```diff
@@ -503,23 +516,111 @@ fn column_ordering_and_projection() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
-#[test]
-fn predicate_on_number() -> Result<(), Box<dyn std::error::Error>> {
-    let expected = vec![
+// get the basic_partitioned table for a set of expected numbers
+fn table_for_numbers(nums: Vec<u32>) -> Vec<String> {
+    let mut res: Vec<String> = vec![
         "+---------+--------+",
         "| a_float | number |",
         "+---------+--------+",
-        "| 1.1     | 1      |",
-        "| 2.2     | 2      |",
-        "| 3.3     | 3      |",
-        "+---------+--------+",
+    ]
+    .into_iter()
+    .map(String::from)
+    .collect();
+    for num in nums.iter() {
+        res.push(format!("| {num}.{num}     | {num}      |"));
+    }
+    res.push("+---------+--------+".to_string());
+    res
+}
+
+#[test]
+fn predicate_on_number() -> Result<(), Box<dyn std::error::Error>> {
+    let cases = vec![
+        (
+            Expression::Column("number".to_string()).lt(Expression::Literal(Scalar::Long(4))),
```

Comment:
```suggestion
            Expression::column("number").lt(Expression::literal(4i64)),
```
Code volume can be reduced quite a bit this way... might even be worth a `use Expression::{column, literal}` to allow
```suggestion
            column("number").lt(literal(4i64)),
```

Reply:
You can't `use` struct methods like that, but yeah, just using the function forms is much better